File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        n = len(s)
        mp = defaultdict(int)
        res = 0

        i, j = 0, 0

        while j < n:
            mp[s[j]] += 1

            while (j-i+1) - max(mp.values()) > k:
                mp[s[i]] -= 1
                i += 1

            res = max(res, j-i+1)
            j += 1

        return res

File: Data Structures & Algorithms/longest-palindromic-substring/submission-0.cpp
class Solution {
public:
    bool isPalindrome(string &s, int i, int j)
    {
        if(i >= j)
            return true;

        if(s[i] == s[j])
            return isPalindrome(s, i+1, j-1);
        return false;
    }
    /////////////////////////////////////
    string longestPalindrome(string s) 
    {
        int n=s.size();

        int startIdx = 0;
        int maxLen = 0;
        for(int i=0;i<n;++i)
        {
            for(int j=i;j<n;++j)
            {
                if(isPalindrome(s, i, j))
                {
                    if(j-i+1 > maxLen)
                    {
                        startIdx = i;
                        maxLen = max(maxLen, j-i+1);
                    }
                }
            }
        }
        cout<<startIdx<<maxLen;
        return s.substr(startIdx, maxLen);
    }
};

//recursive
// TC - O(n^3)
